tests/unit/amo/components/TestUserProfile.js

```diff
@@ -290,6 +295,32 @@ describe(__filename, () => {
     expect(root.find(ReportUserAbuse)).toHaveLength(1);
   });
 
+  it('renders two AddonsByAuthorsCard', () => {
+    const root = renderUserProfile();
+
+    expect(root.find(AddonsByAuthorsCard)).toHaveLength(2);
+  });
+
+  it('renders AddonsByAuthorsCard for extensions', () => {
+    const root = renderUserProfile();
+
+    expect(root.find(AddonsByAuthorsCard).at(0))
+      .toHaveProp('addonType', ADDON_TYPE_EXTENSION);
+  });
+
+  it('renders AddonsByAuthorsCard for themes', () => {
+    const root = renderUserProfile();
+
+    expect(root.find(AddonsByAuthorsCard).at(1))
+      .toHaveProp('addonType', ADDON_TYPE_THEME);
+  });
+
+  it('renders no AddonsByAuthorsCard if no user found', () => {
+    const root = renderUserProfile({ params: { username: 'not-loaded' } });
```

Comment:
I know this is using existing code, but I find it confusing that passing in the string `not-loaded` into `params.username` results in no user being found. It's not really obvious that that is the case. Looking back at the code in this test, and also in the component, the whole `params` thing seems confusing to me. Why does `UserProfile` accept a `params` object as a property rather than just accepting a `username` as a property? This seems to result in having to jump through hoops in the test like `syncPropsAndParamsUsername`.

This doesn't block this PR at all, but I am confused by all of this so wouldn't mind understanding a bit better via an explanation.